src/controllers/library.py

```diff
@@ -94,25 +94,24 @@ def search_library_items(self, library_item_title=None, library_item_isbn=None):
             List[str]: List of matching item titles.
         """
         results = []
-        for library_item in self.library_items.values():
+        for library_item in LibraryItemDocument.objects():
             if (library_item_title is None or library_item_title in library_item.title) or \
                     (library_item_isbn is None or library_item_isbn == library_item.isbn):
```

Comment:
You are not using mongo functionalities.
Research if there is a way to move this logic to the DB.